### ISSUE TITLE
Generating TCC count matrices 

### ISSUE THREAD

**Issue by sbooeshaghi** [pachterlab/kb_python#276]
**Describe the issue**
After running kb count and generating gene count matrices I want to generate the TCC matrices without rerunning the alignment. kb count detects that the bus file has been made and runs up until the error listed below:

```
[2024-11-06 18:42:34,015]    INFO [count_nac] Summing matrices into kb_out_nac_isoform/counts_unfiltered/cells_x_tcc.cell.mtx
[2024-11-06 18:42:51,452]   ERROR [main] An exception occurred
Traceback (most recent call last):
  File "/home/sina/.envs/basenv/lib/python3.12/site-packages/kb_python/main.py", line 1618, in main
    COMMAND_TO_FUNCTION[args.command](parser, args, temp_dir=temp_dir)
  File "/home/sina/.envs/basenv/lib/python3.12/site-packages/kb_python/main.py", line 592, in parse_count
    count_nac(
  File "/home/sina/.envs/basenv/lib/python3.12/site-packages/ngs_tools/logging.py", line 62, in inner
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/sina/.envs/basenv/lib/python3.12/site-packages/kb_python/count.py", line 2076, in count_nac
    unfiltered_results[prefix][f'txnames{suffix}'] if tcc
    ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
KeyError: 'txnames'
```

**What is the exact command that was run?**
```
 kb count --verbose --tcc --strand unstranded --sum cell --filter bustools -t 8 -m 16G -x 0,0,16:0,16,26:0,26,75,1,0,75 -w /home/sina/projects/cellatlas/human/data/ERS3861771/onlist.txt -i ../../reference/nac/index.idx -g ../../reference/nac/t2g.txt -c1 ../../reference/nac/spl.t2c.txt -c2 ../../reference/nac/unspl.t2c.txt --h5ad --workflow=nac -o kb_out_nac_isoform/ /home/sina/projects/cellatlas/human/data/ERS3861771/./fastqs/4834STDY7002886_S1_L001_R1_001.fastq.gz /home/sina/projects/cellatlas/human/data/ERS3861771/./fastqs/4834STDY7002886_S1_L001_R2_001.fastq.gz
```

**Command output (with `--verbose` flag)**
```
$ kb count --verbose --tcc --strand unstranded --sum cell --filter bustools -t 8 -m 16G -x 0,0,16:0,16,26:0,26,75,1,0,75 -w /home/sina/projects/cellatlas/human/data/ERS3861771/onlist.txt -i ../../reference/nac/index.idx -g ../../reference/nac/t2g.txt -c1 ../../reference/nac/spl.t2c.txt -c2 ../../reference/nac/unspl.t2c.txt --h5ad --workflow=nac -o kb_out_nac_isoform/ /home/sina/projects/cellatlas/human/data/ERS3861771/./fastqs/4834STDY7002886_S1_L001_R1_001.fastq.gz /home/sina/projects/cellatlas/human/data/ERS3861771/./fastqs/4834STDY7002886_S1_L001_R2_001.fastq.gz
[2024-11-06 19:07:52,270]   DEBUG [main] Printing verbose output
[2024-11-06 19:07:54,474]   DEBUG [main] kallisto binary located at /home/sina/.envs/basenv/lib/python3.12/site-packages/kb_python/bins/linux/kallisto/kallisto
[2024-11-06 19:07:54,474]   DEBUG [main] bustools binary located at /home/sina/.envs/basenv/lib/python3.12/site-packages/kb_python/bins/linux/bustools/bustools
[2024-11-06 19:07:54,474]   DEBUG [main] Creating `kb_out_nac_isoform/tmp` directory
[2024-11-06 19:07:54,475]   DEBUG [main] Namespace(list=False, command='count', tmp=None, keep_tmp=False, verbose=True, i='../../reference/nac/index.idx', g='../../reference/nac/t2g.txt', x='0,0,16:0,16,26:0,26,75,1,0,75', o='kb_out_nac_isoform/', num=False, w='/home/sina/projects/cellatlas/human/data/ERS3861771/onlist.txt', r=None, t=8, m='16G', strand='unstranded', inleaved=False, genomebam=False, aa=False, gtf=None, chromosomes=None, workflow='nac', em=False, mm=False, tcc=True, filter='bustools', filter_threshold=None, c1='../../reference/nac/spl.t2c.txt', c2='../../reference/nac/unspl.t2c.txt', overwrite=False, dry_run=False, batch_barcodes=False, loom=False, h5ad=True, loom_names='barcode,target_name', sum='cell', cellranger=False, gene_names=False, N=None, report=False, no_inspect=False, kallisto='/home/sina/.envs/basenv/lib/python3.12/site-packages/kb_python/bins/linux/kallisto/kallisto', bustools='/home/sina/.envs/basenv/lib/python3.12/site-packages/kb_python/bins/linux/bustools/bustools', no_validate=False, no_fragment=False, parity=None, fragment_l=None, fragment_s=None, bootstraps=None, matrix_to_files=False, matrix_to_directories=False, fastqs=['/home/sina/projects/cellatlas/human/data/ERS3861771/./fastqs/4834STDY7002886_S1_L001_R1_001.fastq.gz', '/home/sina/projects/cellatlas/human/data/ERS3861771/./fastqs/4834STDY7002886_S1_L001_R2_001.fastq.gz'])
[2024-11-06 19:07:57,230]    INFO [count_nac] Skipping kallisto bus because output files already exist. Use the --overwrite flag to overwrite.
[2024-11-06 19:07:57,230]    INFO [count_nac] Sorting BUS file kb_out_nac_isoform/output.bus to kb_out_nac_isoform/tmp/output.s.bus
[2024-11-06 19:07:57,230]   DEBUG [count_nac] bustools sort -o kb_out_nac_isoform/tmp/output.s.bus -T kb_out_nac_isoform/tmp -t 8 -m 16G kb_out_nac_isoform/output.bus
[2024-11-06 19:08:12,954]   DEBUG [count_nac] partition time: 1.9757s
[2024-11-06 19:08:15,859]   DEBUG [count_nac] all fits in buffer
[2024-11-06 19:08:20,165]   DEBUG [count_nac] Read in 91421018 BUS records
[2024-11-06 19:08:20,165]   DEBUG [count_nac] reading time 0.74954s
[2024-11-06 19:08:20,165]   DEBUG [count_nac] sorting time 23.7193s
[2024-11-06 19:08:20,166]   DEBUG [count_nac] writing time 3.20214s
[2024-11-06 19:08:22,468]    INFO [count_nac] Inspecting BUS file kb_out_nac_isoform/tmp/output.s.bus
[2024-11-06 19:08:22,468]   DEBUG [count_nac] bustools inspect -o kb_out_nac_isoform/inspect.json -w /home/sina/projects/cellatlas/human/data/ERS3861771/onlist.txt kb_out_nac_isoform/tmp/output.s.bus
[2024-11-06 19:08:27,777]    INFO [count_nac] Correcting BUS records in kb_out_nac_isoform/tmp/output.s.bus to kb_out_nac_isoform/tmp/output.s.c.bus with on-list /home/sina/projects/cellatlas/human/data/ERS3861771/onlist.txt
[2024-11-06 19:08:27,777]   DEBUG [count_nac] bustools correct -o kb_out_nac_isoform/tmp/output.s.c.bus -w /home/sina/projects/cellatlas/human/data/ERS3861771/onlist.txt kb_out_nac_isoform/tmp/output.s.bus
[2024-11-06 19:08:28,680]   DEBUG [count_nac] Found 737280 barcodes in the on-list
[2024-11-06 19:08:35,288]   DEBUG [count_nac] Processed 64033044 BUS records
[2024-11-06 19:08:35,289]   DEBUG [count_nac] In on-list = 61245661
[2024-11-06 19:08:35,289]   DEBUG [count_nac] Corrected    = 849279
[2024-11-06 19:08:35,289]   DEBUG [count_nac] Uncorrected  = 1938104
[2024-11-06 19:08:37,491]    INFO [count_nac] Sorting BUS file kb_out_nac_isoform/tmp/output.s.c.bus to kb_out_nac_isoform/output.unfiltered.bus
[2024-11-06 19:08:37,491]   DEBUG [count_nac] bustools sort -o kb_out_nac_isoform/output.unfiltered.bus -T kb_out_nac_isoform/tmp -t 8 -m 16G kb_out_nac_isoform/tmp/output.s.c.bus
[2024-11-06 19:08:51,813]   DEBUG [count_nac] partition time: 0.523592s
[2024-11-06 19:08:55,118]   DEBUG [count_nac] all fits in buffer
[2024-11-06 19:08:58,823]   DEBUG [count_nac] Read in 62094940 BUS records
[2024-11-06 19:08:58,824]   DEBUG [count_nac] reading time 0.493758s
[2024-11-06 19:08:58,824]   DEBUG [count_nac] sorting time 14.4514s
[2024-11-06 19:08:58,824]   DEBUG [count_nac] writing time 2.61074s
[2024-11-06 19:09:01,527]    INFO [count_nac] Generating count matrix kb_out_nac_isoform/counts_unfiltered/cells_x_tcc from BUS file kb_out_nac_isoform/output.unfiltered.bus
[2024-11-06 19:09:01,527]   DEBUG [count_nac] bustools count -o kb_out_nac_isoform/counts_unfiltered/cells_x_tcc -g ../../reference/nac/t2g.txt -e kb_out_nac_isoform/matrix.ec -t kb_out_nac_isoform/transcripts.txt -s ../../reference/nac/unspl.t2c.txt --multimapping --umi-gene kb_out_nac_isoform/output.unfiltered.bus
[2024-11-06 19:25:24,241]   DEBUG [count_nac] kb_out_nac_isoform/counts_unfiltered/cells_x_tcc.mature.mtx passed validation
[2024-11-06 19:25:24,515]   DEBUG [count_nac] kb_out_nac_isoform/counts_unfiltered/cells_x_tcc.nascent.mtx passed validation
[2024-11-06 19:25:24,686]   DEBUG [count_nac] kb_out_nac_isoform/counts_unfiltered/cells_x_tcc.ambiguous.mtx passed validation
[2024-11-06 19:25:24,687]    INFO [count_nac] Summing matrices into kb_out_nac_isoform/counts_unfiltered/cells_x_tcc.cell.mtx
[2024-11-06 19:25:36,883]    INFO [count_nac] Summing matrices into kb_out_nac_isoform/counts_unfiltered/cells_x_tcc.cell.mtx
[2024-11-06 19:25:54,587]   ERROR [main] An exception occurred
Traceback (most recent call last):
  File "/home/sina/.envs/basenv/lib/python3.12/site-packages/kb_python/main.py", line 1618, in main
    COMMAND_TO_FUNCTION[args.command](parser, args, temp_dir=temp_dir)
  File "/home/sina/.envs/basenv/lib/python3.12/site-packages/kb_python/main.py", line 592, in parse_count
    count_nac(
  File "/home/sina/.envs/basenv/lib/python3.12/site-packages/ngs_tools/logging.py", line 62, in inner
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/sina/.envs/basenv/lib/python3.12/site-packages/kb_python/count.py", line 2076, in count_nac
    unfiltered_results[prefix][f'txnames{suffix}'] if tcc
    ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
KeyError: 'txnames'
[2024-11-06 19:25:54,590]   DEBUG [main] Removing `kb_out_nac_isoform/tmp` directory
```

**Comment by Yenaled** [comment]
What version? I think this is fixed in the latest version 0.29.1

**Comment by sbooeshaghi** [comment]
v0.28.2. error is resolved after updating to 0.29.1
